scubagoggles/main.py

```diff
@@ -6,119 +6,399 @@
+    parser.add_argument('--outputfoldername',
+                        default = output_folder,
+                        metavar = '<name>',
+                        help = help_msg)
+
+    provider_filename = default_file_names.provider_output_name
+    help_msg = ('The name of the Provider output json in --outputpath. '
+                f'Defaults to {provider_filename}.')
+    parser.add_argument('--outputproviderfilename',
+                        default = provider_filename,
+                        metavar = '<name>',
+                        help = help_msg)
+
+    rego_filename = default_file_names.rego_output_name
+    help_msg = ('The name of the Rego output json in --outputpath. '
+                f'Defaults to {rego_filename}.')
+    parser.add_argument('--outputregofilename',
+                        default = rego_filename,
+                        metavar = '<name>',
+                        help = help_msg)
+
+    report_filename = default_file_names.report_output_name
+    help_msg = ('The name of the main html file homepage created in '
+                '--outputpath. Defaults to '
+                f'{report_filename}.')
+    parser.add_argument('--outputreportfilename',
+                        default = report_filename,
+                        metavar = '',
+                        help = help_msg)
+
+    help_msg = ('This switch suppresses automatically launching a web '
+                'browser to open the html report output and the loading '
+                'bar output.')
+    parser.add_argument('--quiet', action = 'store_true', help = help_msg)
+
+    help_msg = 'This switch is used to print debugging information for OPA.'
+    parser.add_argument('--debug', action = 'store_true', help = help_msg)
+
+    group = parser.add_argument_group('Cached Mode options')
+
+    help_msg = ('This switch when added will run in the tool in '
+                '"RunCached mode". When combined with --skipexport allows '
+                'the user to skip authentication and provider export.')
+    group.add_argument('--runcached',
+                       action = 'store_true',
+                       help = help_msg)
+
+    help_msg = ('This switch when added will skip the provider export. To be '
+                'used in conjunction with --runcached.')
+    group.add_argument('--skipexport',
+                       action = 'store_true',
+                       help = help_msg)
+
+
+def get_opa_args(parser: argparse.ArgumentParser, user_config: UserConfig):
+
+    """Adds the arguments for the "get OPA" parser.
+
+    :param argparse.ArgumentParser parser: argparse object
+    """
+
+    parser.set_defaults(dispatch = getopa)
+
+    parser.set_defaults(user_config = user_config)
+
+    parser.add_argument('--check',
+                        '-c',
+                        default = False,
+                        action = 'store_true',
+                        help = 'Check hash code after download')
```

Comment:
Change this to something like "skipcheck", defaulting to False. Following secure by design principles, the secure option needs to be the default.

Reply:
done